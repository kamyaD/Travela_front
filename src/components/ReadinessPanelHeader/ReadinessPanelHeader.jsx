import React, { PureComponent } from 'react';
import  PropTypes from 'prop-types';
import PageHeader from '../PageHeader';
import './ReadinessHeader.scss';

class ReadinessPanelHeader extends PureComponent {

  openVisaModal = () => {
    const { openModal } = this.props;
    openModal(true, 'add visa');
  };

  render() {
    return (
      <PageHeader
        title="Travel Readiness"
        actionBtn="Add Visa"
        openModal={this.openVisaModal}
      />
    );
  }
}

ReadinessPanelHeader.propTypes = {
  openModal: PropTypes.func.isRequired,
};

export default ReadinessPanelHeader;
