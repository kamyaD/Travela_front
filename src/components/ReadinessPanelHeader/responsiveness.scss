.readiness-panel-header {
  @media (max-width: 839px) {
    height: 150.9px;
    width: 100%;
  }

  .open-readiness {
    @media (max-width: 479px){
      top:70px
    }
  }
}