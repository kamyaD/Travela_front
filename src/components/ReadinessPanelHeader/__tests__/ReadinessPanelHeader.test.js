import React from 'react';
import ReadinessPanelHeader from '../ReadinessPanelHeader';

const props = {
  openModal: jest.fn()
};
describe('<ReadinessPanelHeader />', () => {
  const wrapper = mount(
    <ReadinessPanelHeader {...props} />
  );
  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('calls open modal', () => {
    expect(wrapper.find('.action-btn').length).toEqual(1);
    wrapper.find('.action-btn').simulate('click');
    expect(props.openModal).toHaveBeenCalled();
  });
});
