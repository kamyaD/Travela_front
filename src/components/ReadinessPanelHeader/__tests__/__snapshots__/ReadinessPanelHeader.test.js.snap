// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReadinessPanelHeader /> renders without crashing 1`] = `
<div
  className="PageHeader"
>
  <div>
    <span
      className="title"
    >
      Travel Readiness
    </span>
  </div>
  <div>
    <button
      className="action-btn btn-new-request"
      onClick={[Function]}
      type="button"
    >
      Add Visa
    </button>
  </div>
</div>
`;
